components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/impl/APIUsageStatisticsRestClientImpl.java

```diff
@@ -2288,6 +2291,142 @@ public void deployArtifacts(String url, String user, String pass) throws Excepti
+                if (api.getContext().equals(usageEntry.getContext()) && api.getId().getApiName().equals(apiName)) {
+                    PerUserAPIUsageDTO usageDTO = usageByUsername.get(usageEntry.getUsername());
+                    if (usageDTO != null) {
+                        usageDTO.setCount(usageDTO.getCount() + usageEntry.getRequestCount());
+                    } else {
+                        usageDTO = new PerUserAPIUsageDTO();
+                        usageDTO.setUsername(usageEntry.getUsername());
+                        usageDTO.setCount(usageEntry.getRequestCount());
+                        usageByUsername.put(usageEntry.getUsername(), usageDTO);
+                    }
+                    break;
+                }
+            }
+        }
+
+        return getTopEntries(new ArrayList<PerUserAPIUsageDTO>(usageByUsername.values()), limit);
+    }
+
+    /**
+     * return list of api usage for a particular api and version
+     *
+     * @param providerName API provider name
+     * @param apiName      Name of the API
+     * @param limit        Number of sorted entries to return
+     * @return a List of PerUserAPIUsageDTO objects - Possibly empty
+     * @throws org.wso2.carbon.apimgt.usage.client.exception.APIMgtUsageQueryServiceClientException on error
+     */
+    @Override
+    public List<PerUserAPIUsageDTO> getUsageBySubscribers(String providerName, String apiName, String apiVersion,
+            int limit) throws APIMgtUsageQueryServiceClientException {
+
+        Collection<APIUsageByUser> usageData = getUsageOfAPI(apiName, apiVersion);
+        Map<String, PerUserAPIUsageDTO> usageByUsername = new TreeMap<String, PerUserAPIUsageDTO>();
+        List<API> apiList = getAPIsByProvider(providerName);
+        for (APIUsageByUser usageEntry : usageData) {
+            for (API api : apiList) {
+                if (api.getContext().equals(usageEntry.getContext()) &&
+                        api.getId().getApiName().equals(apiName) &&
+                        api.getId().getVersion().equals(apiVersion) &&
+                        apiVersion.equals(usageEntry.getApiVersion())) {
+
+                    PerUserAPIUsageDTO usageDTO = usageByUsername.get(usageEntry.getUsername());
+                    if (usageDTO != null) {
+                        usageDTO.setCount(usageDTO.getCount() + usageEntry.getRequestCount());
+                    } else {
+                        usageDTO = new PerUserAPIUsageDTO();
+                        usageDTO.setUsername(usageEntry.getUsername());
+                        usageDTO.setCount(usageEntry.getRequestCount());
+                        usageByUsername.put(usageEntry.getUsername(), usageDTO);
+                    }
+                    break;
+                }
+            }
+        }
+
+        return getTopEntries(new ArrayList<PerUserAPIUsageDTO>(usageByUsername.values()), limit);
+    }
+
+    private List<APIUsageByUser> getUsageOfAPI(String apiName, String apiVersion)
```

Comment:
fixed in commit: https://github.com/wso2/carbon-apimgt/commit/0f41b03ec368c33c381cc33a069b5ac692744050